nomad/stream/ndjson.go

```diff
@@ -0,0 +1,105 @@
+package stream
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+var (
+	// NDJsonHeartbeat is the NDJson to send as a heartbeat
+	// Avoids creating many heartbeat instances
+	NDJsonHeartbeat = &structs.NDJson{Data: []byte("{}\n")}
```

Comment:
👍  - also, do we need to insert the newline here?

Reply:
we do to adhere to our new line delimited json, we could json encode an empty struct which would include it as well but I just opted for writing out the bytes

Reply:
removed!